### PR TITLE
fix(ingest/superset): support superset v2

### PR DESCRIPTION
**Pull request by hsheth2**
Needed as we update our own test superset ingestion.

It turned out that the trailing `/` in the API requests made all the difference. Also adds better error handling around API responses.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/superset.py [hunk 1]
```diff
@@ -251,15 +256,20 @@ def emit_dashboard_mces(self) -> Iterable[MetadataWorkUnit]:
 
         while current_dashboard_page * PAGE_SIZE <= total_dashboards:
             dashboard_response = self.session.get(
-                f"{self.config.connect_uri}/api/v1/dashboard",
+                f"{self.config.connect_uri}/api/v1/dashboard/",
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/superset.py:259]: ???

File: metadata-ingestion/src/datahub/ingestion/source/superset.py [hunk 2]
```diff
@@ -352,9 +362,13 @@ def emit_chart_mces(self) -> Iterable[MetadataWorkUnit]:
 
         while current_chart_page * PAGE_SIZE <= total_charts:
             chart_response = self.session.get(
-                f"{self.config.connect_uri}/api/v1/chart",
+                f"{self.config.connect_uri}/api/v1/chart/",
                 params=f"q=(page:{current_chart_page},page_size:{PAGE_SIZE})",
             )
+            if chart_response.status_code != 200:
+                logger.warning(f"Failed to get chart data: {chart_response.text}")
+            chart_response.raise_for_status()
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/superset.py:370]: will this break entire connector?

Comment (jjoyce0510) [reply metadata-ingestion/src/datahub/ingestion/source/superset.py:370]: (or just skip the chart)

File: metadata-ingestion/src/datahub/ingestion/source/superset.py [hunk 3]
```diff
@@ -251,15 +256,20 @@ def emit_dashboard_mces(self) -> Iterable[MetadataWorkUnit]:
...
             dashboard_response = self.session.get(
-                f"{self.config.connect_uri}/api/v1/dashboard",
+                f"{self.config.connect_uri}/api/v1/dashboard/",
                 params=f"q=(page:{current_dashboard_page},page_size:{PAGE_SIZE})",
             )
+            if dashboard_response.status_code != 200:
+                logger.warning(
+                    f"Failed to get dashboard data: {dashboard_response.text}"
+                )
+            dashboard_response.raise_for_status()
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/superset.py:266]: will this break full connector or just for these dashboards?

File: metadata-ingestion/src/datahub/ingestion/source/superset.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/superset.py:259]: Yep - that's what caused the incompatibility with superset v2. I tested with v1 as well and that seems to still work

File: metadata-ingestion/src/datahub/ingestion/source/superset.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/superset.py:370]: This is the call to get the list of charts. It still breaks the entire connector, as was the behavior before, but just makes the error message more clear

File: metadata-ingestion/src/datahub/ingestion/source/superset.py (hunk 3 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/superset.py:266]: same as above

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**